### PR TITLE
Implement capability to skip server installation

### PR DESCRIPTION
**Pull request by AnnaWyrwal**
New capability 'skipServerInstallation' is introduced in
order to allow user to skip UIAutomator2Server installation phase.
The installation and validation of currently installed versions can be
omitted if user is sure that applications in proper versions are already installed.
It allows for huge performance improvement (even about 50% in some cases).

### CODE REVIEW

File: lib/uiautomator2.js [hunk 1]
```diff
@@ -129,9 +129,12 @@ class UiAutomator2Server {
     // kill any uiautomator existing processes
     await this.killUiAutomatorOnDevice();
 
-    logger.info(`Starting uiautomator2 server ${serverVersion}`);
-
-    logger.info(`Using UIAutomator2 server from '${apkPath}' and test from '${testApkPath}'`);
+    if (!caps.skipUiAutomator2ServerInstallation) {
```
Comment (mykola-mokhnach) [inline lib/uiautomator2.js]: positive conditions are easier to read than negative

File: lib/desired-caps.js [hunk 2]
```diff
@@ -32,6 +32,9 @@ let uiautomatorCapConstraints = {
   mjpegScreenshotUrl: {
     isString: true,
   },
+  skipUiAutomator2ServerInstallation: {
```
Comment (mykola-mokhnach) [inline lib/desired-caps.js]: Please rename the cap to `skipServerInstallation`

File: lib/driver.js [hunk 3]
```diff
@@ -393,7 +393,11 @@ class AndroidUiautomator2Driver extends BaseDriver {
     });
     this.proxyReqRes = this.uiautomator2.proxyReqRes.bind(this.uiautomator2);
 
-    await this.uiautomator2.installServerApk(this.opts.uiautomator2ServerInstallTimeout);
+    if (!this.opts.skipUiAutomator2ServerInstallation) {
```
Comment (mykola-mokhnach) [inline lib/driver.js]: Please change the condition to a positive one

File: lib/driver.js [hunk 4]
```diff
@@ -393,7 +393,11 @@ class AndroidUiautomator2Driver extends BaseDriver {
...
+      await this.uiautomator2.installServerApk(this.opts.uiautomator2ServerInstallTimeout);
+    } else {
+      logger.debug(`'skipUiAutomator2ServerInstallation' is set. Skipping UiAutomator2 server installation.`);
```
Comment (mykola-mokhnach) [inline lib/driver.js]: `info` level would be better

File: lib/uiautomator2.js [hunk 5]
```diff
@@ -129,9 +129,12 @@ class UiAutomator2Server {
...
+      logger.info(`Starting uiautomator2 server ${serverVersion}`);
```
Comment (mykola-mokhnach) [inline lib/uiautomator2.js]: `UIAutomator2` should be spelled equally everywhere

File: lib/uiautomator2.js [hunk 6]
```diff
@@ -129,9 +129,12 @@ class UiAutomator2Server {
...
+      logger.info(`Using UIAutomator2 server from '${apkPath}' and test from '${testApkPath}'`);
+    } else {
+      logger.info(`'skipUiAutomator2ServerInstallation' is set. Attempt to use UiAutomator2 server from device.`);
```
Comment (mykola-mokhnach) [inline lib/uiautomator2.js]: Attempting ... from the device

**Comment by mykola-mokhnach** [comment]
@KazuCocoa Where we usually add alter documentation for newly added UIA capabilities?

**Comment by KazuCocoa** [comment]
https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/caps.md#uiautomator2-only is for UIA2
@AnnaWyrwal do you have a chance to add the capability in the doc?
If you have no time, I'll do it

**Comment by AnnaWyrwal** [comment]
@KazuCocoa I think I can do it. I'll look into it ASAP. Thanks!

**Pull request merged**